Expensify/App
# Add Crashlytics to Chat

@timszot @AndrewGable will you please review this?

This PR adds Crashlytics to Chat so that we can get reports of crashes that users are experiencing as they happen. This is configured to only report crashes for *release* builds, dev builds will not report crashes. This PR also updates the fastlane file on iOS to upload the dsym from the build output to crashlytics automatically.

### Fixed Issues
Fixes https://github.com/Expensify/Expensify/issues/142274

### Tests
1. Run `npm install` and `cd ios && pod install` to confirm all the dependencies are up to date
2. Run the apps via `npm run ios`/`npm run android` (debug builds) and confirm that the apps build fine
3. Update some JS to cause a crash. I updated `submitForm` in `ReportActionCompose.js` to do `abc.test();` at the beginning of the function.
4. Build the app again and cause the crash, then go to [Crashlytics](https://console.firebase.google.com/u/0/project/expensify-chat/crashlytics/app/ios:com.chat.expensify.chat/issues?state=open&time=last-seven-days&type=crash). Confirm that the crash isn't uploaded.
5. Go into `~/Library/Developer/Xcode/DerivedData/ReactNativeChat-bkiuyhfpclvjiabklrsfbkppdtbh/Build/Products/Release-iphonesimulator/` (replace your simluator name) and delete the `Chat.app.dSYM` file if it exists. This will allow us to overwrite it on the next build, so we can upload it to Firebase
5. Build a release version of the app to your device/emulator
    1. On Android, do this by running `react-native run-android --variant=release`
    2. On iOS, you can do this by following [these instructions](https://github.com/Expensify/ReactNativeChat#local-production-build-the-ios-app). If you have issues with profiles, you can also update the Scheme to run with the Release build (follow [these instructions](https://stackoverflow.com/a/18038359/2509962)), just be sure to uncheck `Debug Executable`, then run the build normally to a simulator.
6. For iOS, we need to upload the dsym so that Crashlytics can register the crashes. To do this, run the following from `ReactNativeChat`, be sure to update your simulator name (after `DerivedData`)
```
~/Expensidev/ReactNativeChat/iOS/pods/FirebaseCrashlytics/upload-symbols -gsp ios/GoogleService-Info.plist -p ios  ~/Library/Developer/Xcode/DerivedData/ReactNativeChat-bkiuyhfpclvjiabklrsfbkppdtbh/Build/Products/Release-iphonesimulator/Chat.app.dSYM
```
7. Go back to the [Crashlytics console](https://console.firebase.google.com/u/0/project/expensify-chat/crashlytics/app/ios:com.chat.expensify.chat/issues?state=open&time=last-seven-days&type=crash) and confirm that your crashes show up. **Note that for iOS this seems to take a few minutes (up to 10), but on Android they were posted within 2 minutes.**
7. Confirm you can run the web app and desktop app without any issues
